x-pack/plugins/cases/server/client/cases/get.ts

```diff
@@ -348,3 +351,37 @@ export async function getReporters(
     throw createCaseError({ message: `Failed to get reporters: ${error}`, error, logger });
   }
 }
+
+/**
+ * Retrieves the categories from all the cases.
+ */
+export async function getCategories(
+  params: AllCategoriesFindRequest,
+  clientArgs: CasesClientArgs
+): Promise<string[]> {
+  const {
+    unsecuredSavedObjectsClient,
+    services: { caseService },
+    logger,
+    authorization,
+  } = clientArgs;
+
+  try {
+    const queryParams = decodeWithExcessOrThrow(AllCategoriesFindRequestRt)(params);
+
+    const { filter: authorizationFilter } = await authorization.getAuthorizationFilter(
+      Operations.findCases
```

Comment:
I think it might actually be a bug that `getTags` uses `Operations.findCases` explicitly. How about we switch tags to use `Operations.getTags` and then create a new operation for categories that looks like this:

```
  [ReadOperations.GetCategories]: {
    ecsType: EVENT_TYPES.access,
    name: ACCESS_CASE_OPERATION,
    action: 'case_categories_get',
    verbs: accessVerbs,
    docType: 'cases',
    savedObjectType: CASE_SAVED_OBJECT,
  },
```

in here: x-pack/plugins/cases/server/authorization/index.ts

We'll need to add `ReadOperations.GetCategories` too.

The distinction here is that the authorization logs will show that `find cases` being called where we really want something saying it was trying to get the categories.